### DISCUSSION TITLE
Strongswan not accepting %any as rightid?

### DISCUSSION THREAD

**Discussion by cosmo84** [strongswan/strongswan#488] [Q&A]
Hi all,

I searched the documentation and some other issues already opened, but I don't find the last answer to my situation. 

I configure a S2S VPN between Strongswan (version 5.7.1) and a Check Point firewall. Certificates are used for authentication. 

The authentication of Strongswan on Checkpoint side is successful:

authentication of 'C=Xxxxxx' (myself) successful

Then it runs into an error concerning the ID not matching:

received end entity cert "O=xxxxxxxxx VPN Certificate"
IDir '172.16.0.2' does not match to 'O=xxxxxx'

So in the config I set rightid=172.16.0.2, but it is not working. Then I set rightid=%any, but also this is not working. 

My understanding is that %any tells Strongswan to accept any ID coming from the other side, but then why is it not working?

Any help is appreciated. 

Many thanks,

Stephan

**Comment by tobiasbrunner** [comment]
> So in the config I set rightid=172.16.0.2, but it is not working. Then I set rightid=%any, but also this is not working.

Did you configure `rightcert` too? If so, you can only set `rightid` to a value contained in the certificate (subject DN or subjectAltName) otherwise it will default to the subject DN (check the log for something like `id ... not confirmed by certificate...`). Note that even if you don't configure `rightcert` and set `rightid=%any`, the identity the peer sends (in this case its IP address) has to be confirmed by the certificate (i.e. it has to match a SAN or the subject DN) or the authentication will fail.

**Reply by cosmo84** [reply]
I made another test, putting the full DN as rightid, but still the same error

rightcert=cp700Cert.der
rightid="CN=00:1C:xxxx VPN Certificate,O=00:xxxx


IDir '172.16.0.2' does not match to 'O=00:xxx

No other errors in the log. 

Thanks!!

**Reply by tobiasbrunner** [reply]
Obviously, because the other peer sends the same identity (its private IP address).

**Reply by cosmo84** [reply]
Okay, but then it should work with putting this IP as rightid? But it doesn't.

**Reply by tobiasbrunner** [reply]
No, as I wrote, it won't work if the configured identity is not contained in the certificate.

**Reply by cosmo84** [reply]
No way to ignore this?

Seems to be a bad behavior on Check Point side that they send the IP instead of the DN.

**Reply by tobiasbrunner** [reply]
> No way to ignore this?

No.

> Seems to be a bad behavior on Check Point side that they send the IP instead of the DN.

That's fine as long as the certificate matches.
